src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -251,6 +250,7 @@ public IndexShard(ShardId shardId, IndexSettingsService indexSettingsService, In
         logger.debug("state: [CREATED]");
 
         this.checkIndexOnStartup = indexSettings.get("index.shard.check_on_startup", "false");
+        this.translog = newTranslog(shardId, indexSettings, indexSettingsService, bigArrays, indexStore);
```

Comment:
I'm kind of concerned about the Guice interaction since this is an injected constructor if this call throws an exception. I know Guice freaks out if an exception is thrown in a constructor (for instance, if it couldn't create the translog file due to disk full or something)